test/view/analysis/analysis_screen_test.dart

```diff
@@ -92,13 +99,160 @@ void main() {
       await tester.tap(find.byKey(const Key('goto-previous')));
       await tester.pumpAndSettle();
 
-      final currentMove = find.widgetWithText(InlineMove, 'Kc1');
+      final currentMove = find.textContaining('Kc1');
       expect(currentMove, findsOneWidget);
       expect(
-        tester.widgetList<InlineMove>(currentMove).any((e) => e.isCurrentMove),
+        tester
+            .widget<InlineMove>(
+              find.ancestor(
+                of: currentMove,
+                matching: find.byType(InlineMove),
+              ),
+            )
+            .isCurrentMove,
         isTrue,
       );
     });
+
+    group('Analysis Tree View', () {
+      Future<void> buildTree(
+        WidgetTester tester,
+        String pgn,
+      ) async {
+        final app = await makeTestProviderScopeApp(
+          tester,
+          home: AnalysisScreen(
+            pgnOrId: pgn,
+            options: const AnalysisOptions(
+              isLocalEvaluationAllowed: false,
+              variant: Variant.standard,
+              orientation: Side.white,
+              opening: opening,
+              id: standaloneAnalysisId,
+            ),
+          ),
+        );
+
+        await tester.pumpWidget(app);
+      }
+
+      Text parentText(WidgetTester tester, String move) {
+        return tester.widget<Text>(
+          find.ancestor(
+            of: find.text(move),
+            matching: find.byType(Text),
+          ),
+        );
+      }
+
+      void expectSameLine(WidgetTester tester, Iterable<String> moves) {
+        final line = parentText(tester, moves.first);
+
+        for (final move in moves.skip(1)) {
+          final moveText = find.text(move);
+          expect(moveText, findsOneWidget);
+          expect(
+            parentText(tester, move),
+            line,
+          );
+        }
+      }
+
+      void expectDifferentLines(
+        WidgetTester tester,
+        List<String> moves,
+      ) {
+        for (int i = 0; i < moves.length; i++) {
+          for (int j = i + 1; j < moves.length; j++) {
+            expect(
+              parentText(tester, moves[i]),
+              isNot(parentText(tester, moves[j])),
+            );
+          }
+        }
+      }
+
+      testWidgets('displays short sideline as inline', (tester) async {
+        await buildTree(tester, '1. e4 e5 (1... d5 2. exd5) 2. Nf3 *');
+
+        final mainline = find.ancestor(
+          of: find.text('1. e4'),
+          matching: find.byType(Text),
+        );
+        expect(mainline, findsOneWidget);
+
+        expectSameLine(tester, ['1. e4', 'e5', '1… d5', '2. exd5', '2. Nf3']);
+      });
+
+      testWidgets('displays long sideline on its own line', (tester) async {
+        await buildTree(
+          tester,
+          '1. e4 e5 (1... d5 2. exd5 Qxd5 3. Nc3 Qd8 4. d4 Nf6) Nc3 *',
```

Comment:
Looks like it's missing the `2. ` before `Nc3`.

Reply:
ah indeed, but looks like our PGN parsing logic is fine with that? But I changed it now